File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int sm=0;
        for (int n: nums){
            sm+=n;
        }
        if (sm==x) return nums.length;
        int t=sm-x, left=0, curr=0, maxi=0;
        for (int right=0;right<nums.length;right++){
            curr+=nums[right];
            while (left<=right && curr>t){
                curr-=nums[left];
                left++;
            }
            if (curr==t){
                maxi=Math.max(maxi,right-left+1);
            }
        }
        return maxi>0 ? nums.length-maxi : -1;
    }
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        t,n = sum(nums)-x, len(nums)
        if t == 0:
            return n
        left, curr, maxi = 0,0,0
        for right, val in enumerate(nums):
            curr += val
            while left<=right and curr>t:
                curr -= nums[left]
                left += 1
            if t == curr:
                maxi = max(maxi, right-left+1)
        return n-maxi if maxi else -1